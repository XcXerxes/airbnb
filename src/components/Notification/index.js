import React, {Component} from 'react'
import colors from '../../styles/colors'
import PropTypes from 'prop-types'
import Icon from 'react-native-vector-icons/FontAwesome'
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Animated,
  Easing
} from 'react-native'

export default class Notification extends Component {
  constructor (props) {
    super(props)
    this.state = {
      positionValue: new Animated.Value(60)
    }
  }
  animateNotification = (value) => {
    const {positionValue} = this.state
    Animated.timing(
      positionValue,
      {
        toValue: value,
        duration: 400,
        velocity: 3,
        tension: 2,
        friction: 8,
        easing: Easing.easeOutBack
      }
    ).start()
  }
  closeNotification = () => {
    this.props.handleCloseNotification()
  }
  render () {
    const {type, firstLine, secondLine, showNotification} = this.props
    showNotification ? this.animateNotification(0) : this.animateNotification(60)
    const {positionValue} = this.state
    return (
      <Animated.View 
      style={[{transform: [{translateY: positionValue}]}, styles.wrapper]}>
        <View style={styles.notificationWrapper}>
          <Text style={styles.errorText}>{type}</Text>
          <Text style={styles.errorMessage}>{firstLine}</Text>
          <Text style={styles.errorMessage}>{secondLine}</Text>
        </View>
        <TouchableOpacity style={styles.closeButton} onPress={this.closeNotification}>
          <Icon name="times" size={20} color={colors.lightGray} />
        </TouchableOpacity>
      </Animated.View>
    )
  }
}

Notification.proptypes = {
  showNotification: PropTypes.bool.isRequired,
  type: PropTypes.string,
  firstLine: PropTypes.string,
  secondLine: PropTypes.string,
  handleCloseNotification: PropTypes.func
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: colors.white,
    height: 60,
    width: '100%',
    padding: 10,
  },
  notificationWrapper: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'flex-start'
  },
  errorText: {
    color: colors.darkOrange,
    marginRight: 5,
    fontSize: 14,
    marginBottom: 2
  },
  errorMessage: {
    marginBottom: 2,
    fontSize: 14,
  },
  closeButton: {
    position: 'absolute',
    right: 10,
    top: 10
  }
})