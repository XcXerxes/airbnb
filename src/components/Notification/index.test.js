import React from 'react'
import Notification from './'

import TestRender from 'react-test-renderer'

const testRender = TestRender.create(<Notification type="success" firstLine="first" secondLine="second" />)
console.log(testRender)
const testInstance = testRender.root
const testTree = testRender.toJSON()

it('should return shot', () => {
  expect(testTree).toMatchSnapshot()
})

// it('should return props', () => {
//   expect(testInstance.findByType(Notification).props.type).toBe('success')
//   expect(testInstance.findByType(Notification).props.firstLine).toBe('first')
//   expect(testInstance.findByType(Notification).props.secondLine).toBe('second')
// })