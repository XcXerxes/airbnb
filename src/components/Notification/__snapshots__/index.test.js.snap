// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return shot 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": 60,
      "padding": 10,
      "transform": Array [
        Object {
          "translateY": 60,
        },
      ],
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#d93900",
          "fontSize": 14,
          "marginBottom": 2,
          "marginRight": 5,
        }
      }
    >
      success
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 2,
        }
      }
    >
      first
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 2,
        }
      }
    >
      second
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": 10,
        "top": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;
